src/components/structures/TimelinePanel.js

```diff
@@ -1023,7 +1020,16 @@ var TimelinePanel = React.createClass({
             var boundingRect = node.getBoundingClientRect();
             if ((allowPartial && boundingRect.top < wrapperRect.bottom) ||
                 (!allowPartial && boundingRect.bottom < wrapperRect.bottom)) {
-                return i;
+                // find last invisible event before the next visible one.
```

Comment:
E.g

ev1
ev2
ev3-hidden

riot-android could set RM to ev3, but riot-web would then set it to ev2, with this the RM would fall onto hidden events even though they aren't in the DOM.